mirdata/mridangam_stroke.py

```diff
@@ -0,0 +1,262 @@
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            tags_open_data=[(self.stroke_name, 'stroke_name')],
+            metadata={
+                'tonic': self.tonic
+            },
+        )
+
+
+def load_audio(audio_path):
+    """Load a Mridangam Stroke Dataset audio file.
+    Args:
+        audio_path (str): path to audio file
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=44100, mono=True)
+
+
+def download(data_home=None, force_overwrite=False, cleanup=True):
+    """Download the Mridangam Stroke Dataset.
+    Args:
+        data_home (str):
+            Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+        force_overwrite (bool):
+            Whether to overwrite the existing downloaded data
+        cleanup (bool):
+            Whether to delete the zip/tar file after extracting.
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    download_utils.downloader(
+        data_home,
+        remotes=REMOTES,
+        info_message=None,
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+
+
+def validate(data_home=None, silence=False):
+    """Validate if the stored dataset is a valid version
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths that file exists in the dataset
+            index but has a different checksum compare to the reference checksum
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    missing_files, invalid_checksums = utils.validator(
+        DATA.index, data_home, silence=silence
+    )
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    """Return track ids
+    Returns:
+        (list): A list of track ids
+    """
+    return list(DATA.index.keys())
+
+
+def load(data_home=None):
+    """Load Mridangam Stroke dataset
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        (dict): {`track_id`: track data}
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    data = {}
+    for key in DATA.index.keys():
+        data[key] = Track(key, data_home=data_home)
+    return data
+
+
+def load_stroke_name(audio_path):
```

Comment:
In general, our convention is that `load_*` methods should be loading content from a file, rather than parsing strings. Can you make these both Track attributes defined in `__init__`? `self.stroke_name = ...` and `self.tonic = ...`

Also, since the filepath can be parsed even if the audio file doesn't exist,  we don't need to check if it exists.

Reply:
Fixed!